test/api/route/typing_test.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:checks/checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/route/messages.dart';
+import 'package:zulip/api/route/typing.dart';
+
+import '../../stdlib_checks.dart';
+import '../fake_api.dart';
+
+void main() {
+  const streamId = 123;
+  const topic = 'topic';
+  const userIds = [101, 102, 103];
+
+  Future<void> checkSetTypingStatus(FakeApiConnection connection, TypingOp op, {
+    required MessageDestination destination,
```

Comment:
nit:
```suggestion
  Future<void> checkSetTypingStatus(FakeApiConnection connection,
    TypingOp op, {
    required MessageDestination destination,
```

else "op" gets a bit lost visually